[BUG] delta_lake_test.py::test_delta_read_column_mapping fails with delta 3.3.0

**Describe the bug**
The test in question fails with delta 3.3.0 as follows when the deletion vector is disabled and the parquet reader type is `PERFILE`. Note that it fails with both column mapping modes of `id` and `name`. 

```
FAILED ../../src/main/python/delta_lake_test.py::test_delta_read_column_mapping[False-name-{'spark.rapids.sql.format.parquet.reader.type': 'PERFILE'}][DATAGEN_SEED=1748987997, TZ=UTC, IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - pyspark.errors.exceptions.captured.AnalysisException: [DELTA_CANNOT_RESOLVE_COLUMN] Can't resolve column col-791428e7-1d65-49a7-ba25-817681c0d77f in root
 |-- a: integer (nullable = true)
 |-- b: string (nullable = true)
 |-- c: string (nullable = true)
 |-- d: integer (nullable = true)
 |-- e: long (nullable = true)
FAILED ../../src/main/python/delta_lake_test.py::test_delta_read_column_mapping[False-id-{'spark.rapids.sql.format.parquet.reader.type': 'PERFILE'}][DATAGEN_SEED=1748987997, TZ=UTC, IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - pyspark.errors.exceptions.captured.AnalysisException: [DELTA_CANNOT_RESOLVE_COLUMN] Can't resolve column col-c016ed19-07bf-4d70-ab89-b61de407880d in root
 |-- a: integer (nullable = true)
 |-- b: string (nullable = true)
 |-- c: string (nullable = true)
 |-- d: integer (nullable = true)
 |-- e: long (nullable = true)
```

**Steps/Code to reproduce bug**
Build the plugin for spark 3.5 and run the following command.

```
PYSP_TEST_spark_jars_packages="io.delta:delta-spark_2.12:3.3.0" PYSP_TEST_spark_sql_extensions="io.delta.sql.DeltaSparkSessionExtension" PYSP_TEST_spark_sql_catalog_spark__catalog="org.apache.spark.sql.delta.catalog.DeltaCatalog" TESTS=delta_lake_test.py::test_delta_read_column_mapping  integration_tests/run_pyspark_from_build.sh -m delta_lake --delta_lake
```

**Expected behavior**
The test should pass with all parameters.

--- a/delta-lake/common/src/main/delta-io/scala/com/nvidia/spark/rapids/delta/GpuDeltaParquetFileFormat.scala
+++ b/delta-lake/common/src/main/delta-io/scala/com/nvidia/spark/rapids/delta/GpuDeltaParquetFileFormat.scala
@@ -36,6 +36,11 @@ trait GpuDeltaParquetFileFormat extends GpuReadParquetFileFormat {
   val columnMappingMode: DeltaColumnMappingMode
   val referenceSchema: StructType
 
+  /**
+   * prepareSchema must only be used for parquet read.
+   * It removes "PARQUET_FIELD_ID_METADATA_KEY" for name mapping mode which address columns by
+   * physical name instead of id.
+   */
   def prepareSchema(inputSchema: StructType): StructType = {
     val schema = DeltaColumnMapping.createPhysicalSchema(
       inputSchema, referenceSchema, columnMappingMode)

--- a/delta-lake/delta-33x/src/main/scala/com/nvidia/spark/rapids/delta/delta33x/GpuDelta33xParquetFileFormat.scala
+++ b/delta-lake/delta-33x/src/main/scala/com/nvidia/spark/rapids/delta/delta33x/GpuDelta33xParquetFileFormat.scala
@@ -72,11 +72,10 @@ case class GpuDelta33xParquetFileFormat(
   }
 
   /**
-   * prepareSchemaForRead must only be used for parquet read.
-   * It removes "PARQUET_FIELD_ID_METADATA_KEY" for name mapping mode which address columns by
-   * physical name instead of id.
+   * This function is overridden as Delta 3.3 has an extra `PARQUET_FIELD_NESTED_IDS_METADATA_KEY`
+   * key to remove from the metadata, which does not exist in earlier versions.
    */
-  def prepareSchemaForRead(inputSchema: StructType): StructType = {
+  override def prepareSchema(inputSchema: StructType): StructType = {
     val schema = DeltaColumnMapping.createPhysicalSchema(
       inputSchema, referenceSchema, columnMappingMode)
     if (columnMappingMode == NameMapping) {
@@ -150,9 +149,9 @@ case class GpuDelta33xParquetFileFormat(
 
     val dataReader = super.buildReaderWithPartitionValuesAndMetrics(
       sparkSession,
-      prepareSchemaForRead(dataSchema),
-      prepareSchemaForRead(partitionSchema),
-      prepareSchemaForRead(requiredSchema),
+      dataSchema,
+      partitionSchema,
+      requiredSchema,
       prepareFiltersForRead(filters),
       options,
       hadoopConf,

